### DISCUSSION TITLE
Can't get route in head meta

### DISCUSSION THREAD

**Discussion by HiddenUndefined** [nuxt/nuxt#18354] [Questions] [answered]
I'll get custom script for generate meta data in head in page, but when I tried get `$route.path`,`route.path`,`this.$route.path`,`this.route.path` – I get error.

Error: `entry.mjs?eb0b:39 Error while mounting app: TypeError: Cannot read properties of undefined (reading '$route')`.

It's my script from `index.vue`

```ts script
<script lang="ts">
export default {
  name: 'Main',
  layout: 'website',
  head: {
    title: 'Project – Info.',
    meta: [
      {
        hid: 'twitterURL',
        property: 'twitter:url',
        // 1-problem: this.$route || this.route || $route || route – doesn't work, return undefined & error.
        content: `${ process.env.APP_BASE_URL }${ this.$route.path }`
      }
    ]
  }
}
</script>
```

Where am I wrong?

**Comment by lucpotage** [comment]
Could you check`vue-router`dependency is included? With an `app.vue` file, it’s not included by default to reduce the bundle size.

**Reply by HiddenUndefined** [reply]
Sorry for stupid question, but how I can include dependencies in `App.vue` or may be I must include `vue-router` in `nuxt.config.ts`?

**Reply by lucpotage** [reply]
Just create a `pages/index.vue` instead of `app.vue`. It should do the job. https://v3.nuxtjs.org/docs/directory-structure/pages

**Reply by HiddenUndefined** [reply]
Oh, it's I created, I have a lot of different pages in `page` directory.

**Reply by lucpotage** [reply]
When you inspect with Nuxt dev tools, do you see a `$route` object?

**Reply by HiddenUndefined** [reply]
I can get `$route` in `template`, but when I tried get `$route` in `script` (`head->meta[]` or `setup()`) – I get error `undefined`.

**Comment by lucpotage** [comment]
Well, I’m new to Vue 3 but I found the answer in Vue Router 4 docs. You cannot access `this` in setup but you can simply use the composition API instead. https://next.router.vuejs.org/guide/advanced/composition-api.html#accessing-the-router-and-current-route-inside-setup

```vue
<script setup>
const route  = useRoute()
</script>
```

**Reply by HiddenUndefined** [reply]
An excellent solution, I don't understand how I could forget about `useRoute()`, but unfortunately this doesn't solve the problem with the inability to access `$route` in `head: { meta: { ... } }` :(

**Reply by lucpotage** [reply]
Answered as a parent comment.

**Reply by HiddenUndefined** [reply]
This is what you need, thanks for the help!
If you do not mind, please make your last answer as a separate comment and I will mark it with a solution.

**Reply by lucpotage** [reply]
Done! Glad it helped.

**Comment by lucpotage** [comment, marked as answer]
Since `head` property does not have access to the component instance, you must rely on the `useMeta` composable instead.

```vue
<script setup>
const route  = useRoute()
useMeta({
  ...
})
</script>
```

**Comment by nathanchase** [comment]
@lucpotage @GeorgeRudenko There are also new Nuxt-provided components for the template you can use:
https://v3.nuxtjs.org/docs/usage/meta-tags/#meta-components
```js
<template>
  <div>
    Hello World
    <Head :lang="dynamic > 50 ? 'en-GB' : 'en-US'">
      <Title>{{ dynamic }} title</Title>
      <Meta name="description" :content="`My page's ${dynamic} description`" />
      <Link rel="preload" href="/test.txt" as="script" />
    </Head>
  </div>
</template>
```
